frontend/src/metabase/query_builder/containers/QueryBuilder.jsx

```diff
@@ -277,7 +277,8 @@ function QueryBuilder(props) {
 
   const handleCreate = useCallback(
     async newQuestion => {
-      const shouldBePinned = newQuestion.isDataset();
+      const shouldBePinned =
+        newQuestion.type() === "model" || newQuestion.type() === "metric";
```

Comment:
Asked in [Slack](https://metaboat.slack.com/archives/C051AG38B2S/p1707908412668569) about it

Reply:
It's correct :heavy_check_mark: